core/lib/ReactCore_Internal.re

```diff
@@ -921,6 +933,303 @@ module Make = (Implementation: HostImplementation) => {
+            traverseRenderedElement(parentHostView, element, f, position),
+          elements,
+        )
+      }
+    and processInstance:
+      type state action elementType.
+        (
+          ~parentHostView: hostView,
+          ~instance: instance(state, action, elementType),
+          ~position: int,
+          ~f: (hostView, int, unit => hostView, int) => hostView
+        ) =>
+        unit =
+      (
+        ~parentHostView,
+        ~instance: instance(state, action, elementType),
+        ~position,
+        ~f,
+      ) => {
+        let {id, component, subElements, instanceSubTree, _} = instance;
+        let newHostRoot =
+          switch (component.elementType) {
+          | React => parentHostView
+          | Host => f(parentHostView, id, subElements.make, position)
+          };
+        traverseRenderedElement(newHostRoot, instanceSubTree, f, 0);
+      };
+
+    let mountInstanceUtil =
+        (implementationFunc, parentHostView, id, make, position) =>
+      switch (getInstance(id)) {
+      | Some(_) => parentHostView
+      | None =>
+        let hostView = make();
+        memoizeInstance(id, hostView);
+        implementationFunc(
+          ~parent=parentHostView,
+          ~child=hostView,
+          ~position,
+        );
+        hostView;
+      };
+
+    let mountInstance = (~parentHostView, ~instance, ~position) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=mountInstanceUtil(Implementation.mountChild),
+      );
+
+    let remountInstanceUtil =
+        (implementationFunc, parentHostView, id, make, position) =>
+      switch (getInstance(id)) {
+      | Some(hostView) =>
+        implementationFunc(
+          ~parent=parentHostView,
+          ~child=hostView,
+          ~position,
+        );
+        hostView;
+      | None => parentHostView
+      };
+
+    let remountInstance = (~parentHostView, ~instance, ~position as _) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=remountInstanceUtil(Implementation.remountChild),
+      );
+
+    let unmountInstanceUtil = (parentHostView, id, _, _) =>
+      switch (getInstance(id)) {
+      | Some(hostView) =>
+        Implementation.unmountChild(~parent=parentHostView, ~child=hostView);
+        hostView;
+      | None =>
+        /* TODO Determine if/when this case is even possible? */
+        print_endline(
+          "Host view instance "
+          ++ string_of_int(id)
+          ++ " wasn't found during rendered element unmount",
+        );
+        parentHostView;
+      };
+
+    let unmountInstance = (~parentHostView, ~instance) =>
+      processInstance(
+        ~parentHostView,
+        ~instance,
+        ~position=0,
+        ~f=unmountInstanceUtil,
+      );
+
+    let rec applyUpdateLogEntry:
+      type state action elementType.
+        (
+          ~subtreeChange: option(UpdateLog.subtreeChange),
```

Comment:
why do we pass subtreeChange here as `option` and then use it only in one place?

Reply:
Because it's only used for `topLevelUpdate` and without it we'd have to duplicate a lot of stuff to account for `` `ContentChanged(subtreeChangeReact)``